### ISSUE TITLE
Offset with performARHitTestWithRay method

### ISSUE THREAD

**Issue by MayeulGarreau** [viromedia/viro#379]
Hi everybody, thank you again for making AR possible with RN, it's amazing!

For my AR app, the use case is to allow users to perform an _AR hit test_ in the ViroARScene thanks to a 2D crosshair and then to render a small ViroSphere at the position where the _AR hit test_ has been done.

Thanks to the ViroARSampleApp exemple and particularly to the `performARHitTestWithRay` method, I almost succeeded to develop this feature.

My problem is that when I'm targeting a point in the scene and then perform the _AR hit test_, a little offset appears between the 2D crosshair and the ViroSphere rendered (see picture below, the offset seems to be always on the left-up direction from the crosshair).

![screenshot2](https://user-images.githubusercontent.com/38918066/43827267-b6d9187e-9af9-11e8-95b3-ffdf2b5d8e20.jpg)

Am I misunderstanding something ? Or it is due to the inherent performance of ARKit/ARCore ?

I didn't insert code on this message since I think it's a methodology problem but if you need , I'll give you some.

Thanks in advance for your help!

Mayeul

**Comment by achuvm** [comment]
Hi @MayeulGarreau,

We aren't aware of any issues using this function.

How are you calculating the ray for your hit test?

Is it possible that your `ViroBox` has some positional offset on it?

A code snippet of how you calculate your ray and how you invoke the `performARHitTestWithRay` would help to debug the issue.

Another way to do this is to use the `performARHitTestWithPoint` function that takes an `x` and `y` screen point: https://docs.viromedia.com/docs/viroarscene

Thanks,

**Comment by MayeulGarreau** [comment]
Hi @achuvm,

Sorry for the delay of my answer! 

Previously, I was calculating the ray for my hit test like this : 

`this.refs["arscene"].getCameraOrientationAsync().then((orientation) => {
    this.refs["arscene"].performARHitTestWithRay(orientation.forward).then((results)=>{
       ...
     })
});`

... but it implied the offset subject of this issue (I verified that there was no problem with `ViroSphere`).

I tried your `performARHitTestWithPoint` workaround, and it perfectly works: thank you so much!

FYI, here is the working snippet:

`    this.refs["arscene"].getCameraOrientationAsync().then((orientation) => {
      this.refs["arscene"].performARHitTestWithPoint((Dimensions.get('window').width * PixelRatio.get())/2, (Dimensions.get('window').height * PixelRatio.get())/2).then((results) => { ...
      })
    }); `

I don't really understand why but the `orientation.forward` vector used by `performARHitTestWithRay` doesn't seem to perfectly 'target' the center of the screen!

Thanks again for your help!

Mayeul
